cmd/pinniped-proxy/src/pinniped.rs

```diff
@@ -131,7 +133,17 @@ async fn call_pinniped_exchange(authorization: &str, k8s_api_server_url: &str, k
         Some(a) => a.to_string(),
         None => authorization.to_string(),
     };
-    let token_creds: Api<TokenCredentialRequest> = Api::namespaced(client.clone(), &pinniped_namespace);
+    
+    let pinniped_version = env::var(PINNIPED_VERSION)?;
```

Comment:
So this is a short-term issue that we have until TMC updates (it may have already, I haven't checked). I was going to suggest that we default `PINNIPED_VERSION` to `0.6.0` so that if it's not set, we assume the future stable behavior rather than erroring if it's not set?

Although I think that would be fine, it may be clearer to use a more specific flag `NAMESPACED_TOKEN_CREDENTIAL_REQUESTS` and default that to false if not present.

But realistically, I don't think we need to support both. Given pinniped's state and our use of it, I think it'll be fine for us to update to the non-namespaced client and just say that 0.6.0 is the minimum supported version. What do you think?

Reply:
I have checked it with TMC folks; good and bad news here:
  1. The future stable behavior is the non-namespaced API calls, so we can just require >=0.6.0. Also, TMC is rolling out the new versions automatically, so we should not be worried about it.
  2. But... they will leverage from the `api-group-suffix` already introduced in 0.5.0. And it does have an impact for us.
 
As per 2., we don't have any option but to allow dynamic API groups (TMC will have a special one, whereas pinniped remains to be "pinniped.dev").
In the current update of this PR I'm introducing some logic to support that. However, we should change the way in which we generate the CRD struct. We cannot derive using a runtime var, so the struct creation has to be more complex.

As far as I have read, we have to create a CustomResource manually and implement some traits manually.
Right now I'm a bit stuck here, let's see tomorrow :P 

https://github.com/clux/kube-rs/blob/master/examples/crd_api.rs
https://docs.rs/kube-derive/0.51.0/kube_derive/derive.CustomResource.html


Now, answering your original response: initially, I'd go for the idea of dropping support for <0.6. However, since we have to change the CRD, one more flag is not a big deal and we, therefore, separate the Pinniped-Proxy's lifecycle from the TMC one.

Reply:
> As per 2., we don't have any option but to allow dynamic API groups (TMC will have a special one, whereas pinniped remains to be "pinniped.dev").
> In the current update of this PR I'm introducing some logic to support that. However, we should change the way in which we generate the CRD struct. We cannot derive using a runtime var, so the struct creation has to be more complex.
> 
> As far as I have read, we have to create a CustomResource manually and implement some traits manually.
> Right now I'm a bit stuck here, let's see tomorrow :P

Yeah, so it's unfortunate (for us) that they re-compile pinniped for tmc with different options, and that prefix is a build-time var for us. Interested to see what we need here, but also happy to jump in if you want.

> 
> https://github.com/clux/kube-rs/blob/master/examples/crd_api.rs
> https://docs.rs/kube-derive/0.51.0/kube_derive/derive.CustomResource.html
> 
> Now, answering your original response: initially, I'd go for the idea of dropping support for <0.6. However, since we have to change the CRD, one more flag is not a big deal and we, therefore, separate the Pinniped-Proxy's lifecycle from the TMC one.

Yeah, OK.